File: Jour14/jour14.py
#Coding:utf-8
#----------------------------------------------------------------------------------------- 

def Fibonacci(n):
	Fp = 1
	Fs = 1
	for i in range(n):
		if i == 0 or i == 1:
			print(f"F_{i} = {Fs}")
		else:
			Fpp = Fp
			Fp  = Fs
			Fs += Fpp
			print(f"F_{i} = {Fs}")

#----------------------------------------------------------------------------------------- 
while 1:
	n = input("Donner n pour afficher les n premier termes de la suite de Fibonacci : ")
	try:
		n = int(n)
		if n < 0:
			print("Erreur :::: n est un entier naturel.")
			continue
	except:
		print("Saisie invalide :::: Ressayer.")
		continue
	else:
		break

Fibonacci(n)

File: Jour14/jour14.c
#include<stdio.h>
//-----------------------------------------------------------------------------------------------
//Fonction qui affiche les n premier termes de la suite de fibonacci

void Fibonacci(int n)
{
	int i,Fpp,Fp = 1, Fs = 1;
	for(i = 0; i < n; i++)
	{
		if(i == 0 || i == 1)
			printf("F_%d = %d\n",i,Fs);
		else
		{
			Fpp = Fp;	//Fpp conserve le terme i-2 de la suite 
			Fp = Fs;	//Fp conserve le terme i-1 de la suite
			Fs += Fpp;	//Fs conserve le terme i de la suite à calculer
			printf("F_%d = %d\n",i,Fs);
		}
	}
}
//-----------------------------------------------------------------------------------------------

int main()
{
	int n;
	do
	{
		printf("Donner n pour afficher les n premiers termes de fibonacci : ");
		scanf("%d",&n);
		if(n < 0)
			printf("Erreur ::::: n est un entier naturel.");
	}while(n < 0);

	Fibonacci(n);

	return 0;
}